src/parser.cpp

```diff
@@ -251,8 +252,17 @@ void Parser::ParserImpl::loadModel(const ModelPtr &model, const std::string &inp
         return;
     } else if (!node->isType("model")) {
         ErrorPtr err = std::make_shared<Error>();
-        err->setDescription("Model root node is of invalid type '" + node->getType() +
-                            "'. A valid CellML root node should be of type 'model'.");
+        if (node->getType() == "model") {
```

Comment:
I am not sure I am following you. There are two signatures for `XmlNode::isType()`:
 - `XmlNode::isType(const char *attributeNamespace, const char *elementName)`; and
 - `XmlNode::isType(const char *elementName)`.

with the latter calling the former with `CELLML_2_0_NS` as its first argument, and we use that second version for everything that is related to CellML itself.

Reply:
Ah, I missed that one, thanks!

Reply:
Wait hold on, if it already checks that it's in the cellml namespace, the next check can never fail?

Reply:
is this resolved? I think the first check is to see if the root node is a `model` in the CellML namespace and then if it is not, it checks if it is a `model` element in any namespace as a potentially more useful error condition than the root node being any arbitrary XML node...

Reply:
For me it should be resolved since @MichaelClerx's "issue" is a non-issue for me, and this for the reason @nickerso gave. Ok, so waiting for @MichaelClerx to confirm that he agrees that this should be resolved, so that either @nickerso or @MichaelClerx can be merge this PR.

Reply:
It's resolved. Find it very confusing that `isType()` checks a namespace and `getType` does not return one though.